bedshift/bedshift.py

```diff
@@ -295,6 +297,41 @@ def _shift(self, row, mean, stdev):
         return row, {0: chrom, 1: start + theshift, 2: end + theshift, 3: 1}
 
 
+    def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter='\t'):
+        self._check_rate(shiftrate)
+        if shiftrate == 0:
+            return 0
+        if len(self.chrom_lens) == 0:
+            _LOGGER.error("chrom.sizes file must be specified when shifting regions")
+            sys.exit(1)
+
+        rows = self.bed.shape[0]
+        num_shift = int(rows * shiftrate)
+        shift_bed = self.read_bed(fp, delimiter=delimiter)
+
+        intersect_regions = self._find_overlap(fp)
+        try:
+            rows2shift = random.sample(list(range(len(intersect_regions))), num_shift)
+        except ValueError:
+            bedname = os.path.basename(self.bedfile_path)
+            shift_file_name = os.path.basename(fp)
+            print("The number of overlapping regions between "+
+                "{} and {} is {} but the shift ratio provided is trying to shift {} regions."\
+                .format(bedname, shift_file_name, len(intersect_regions), num_shift))
+            sys.exit(1)
+
+        shift_dict = {}
```

Comment:
At this point, can you modify the existing `shift` instead of writing a new version? I see what you are trying to do with `_shift`, but I think it's better to keep the existing function as it is. 

(What `_shift` is supposed to do is get one single region specified by `row`, then return a dictionary representing a new region. The old region is intended to be dropped.)